.github/workflows/applications.yml

```diff
@@ -243,6 +243,10 @@ jobs:
       run: |
         ghc --version
         cabal --version
+    - name: Install cabal-plan and confirm its installation
+      run: |
+        cabal install cabal-plan --flags="exe license-report" --constraint="cabal-plan==0.7.3.0" --allow-newer
+        cabal-plan --version
```

Comment:
How long does this take?

Can this be cached or installed via a package manager or a container image? Otherwise this build will run for each workflow run.

Reply:
It takes a couple of minutes. I am not sure what the right way to improve this is. I can't find it in any package managers. We could cache it with a nix build using our own cache.

Reply:
Installing and initializing a nix environment just for that one tool seems overkill to me and is probably also slow.

The cabal-cache binary we wrap into an GitHub actions. It downloads the binary from the release page. It also injects it into the runner cache.

A more lightweight approach could be to check if there binaries on the GitHub release page and, if not, clone it and create (stripped) release binaries. Or just put binaries in an S3 bucket.

Another option would be to put a container image into ghcr.io and run it from there. Whatever is the most convenient.

Reply:
I vaguely remember that cabal itself has support for buidling built-tools, but not sure. If building cabal-plan would be part of the normal build, it would profit from the cabal-cache and dist-cache out of the box.

Reply:
One final idea: instead of calling cabal-plan as external binary, could we just use it as a library and call the required functionality directly? That would simplify the TH code (no external process) and required library functions would be just build as dependencies.

How much of cabal-plan code is actually required for the desired functionality? Maybe we could extract and vendor a minimal implementation?

Reply:
Well, I just noticed that cabal-plan is under GPL-2.0. So we can't link against it or copy code from it.
We could still consider implementing the functionality ourself -- if it's not too much work.